common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/OAuth2Strategy.java

```diff
@@ -217,17 +218,23 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
         headers.put(HttpConstants.HeaderField.CONTENT_TYPE, TOKEN_REQUEST_CONTENT_TYPE);
 
         if (request instanceof MicrosoftTokenRequest) {
+            MicrosoftTokenRequest microsoftTokenRequest = (MicrosoftTokenRequest) request;
             headers.put(
                     AuthenticationConstants.AAD.APP_PACKAGE_NAME,
-                    ((MicrosoftTokenRequest) request).getClientAppName()
+                    microsoftTokenRequest.getClientAppName()
             );
             headers.put(
                     AuthenticationConstants.AAD.APP_VERSION,
-                    ((MicrosoftTokenRequest) request).getClientAppVersion()
+                    microsoftTokenRequest.getClientAppVersion()
             );
-            if (((MicrosoftTokenRequest) request).isPKeyAuthHeaderAllowed()) {
+            if (microsoftTokenRequest.isPKeyAuthHeaderAllowed()) {
                 headers.put(PKEYAUTH_HEADER, PKEYAUTH_VERSION);
             }
+            // Not going to add passkey protocol header until full feature is ready.
+            if (FidoConstants.IS_PASSKEY_SUPPORT_READY
+                    && microsoftTokenRequest.isPasskeyAuthHeaderAllowed()) {
+                headers.put(FidoConstants.PASSKEY_PROTOCOL_HEADER, FidoConstants.PASSKEY_PROTOCOL_HEADER_VALUE);
+            }
```

Comment:
We need the header for /token endpoint as well?

Reply:
I did ask if the header was needed for endpoints other than /authorize, and the response was that the header should be added for other calls involving interactive parameters as well... my plan is to test with the protocol once it's available in a test slice, and if the result is the same regardless of the header being present in /token calls, I'll ask the server team again about this.

Reply:
I'd think we shouldn't need this for /token endpoint since we should be using the RT for auth.

Reply:
Since both of you mentioned this, I'm removing this logic: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2181/commits/8a90970d55796d97a386628c3bd025d82e46f23f